### PR TITLE
Onchain development - Updated Intro to onchain lesson

### PR DESCRIPTION
**Pull request by 0xCipherCoder**
### Summary of Changes

- Updated content to be more understandable
- Added code snippet with latest changes
- Fixed grammar issue and styling

Fixes #

### CODE REVIEW

File: content/courses/onchain-development/intro-to-onchain.md [hunk 1]
```diff
@@ -11,133 +11,192 @@ description:
...
-Each Solana cluster (`mainnet-beta`, `testnet`, `devnet`, `localnet`) is
-effectively a single computer with a globally synchronized state. The programs
-that run on Solana - the ones that create tokens, swap tokens, art marketplaces,
-escrows, market makers, DePIN apps, auctions, retail payments platforms, etc -
-are called **Solana apps**.
+Each Solana cluster (`mainnet-beta`, `testnet`, `devnet`, `localnet`) functions
+as a single computer with a globally synchronized state. The programs that
+operate on Solana—creating tokens, swapping tokens, powering art marketplaces,
+escrows, market makers, DePIN apps, auctions, retail payment platforms, and
+more—are referred to as **Solana apps**.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:30]:
```suggestion
Each Solana cluster (`mainnet-beta`, `testnet`, `devnet`) functions
as a single computer with a globally synchronized state. The programs that
operate on Solana—creating tokens, swapping tokens, powering art marketplaces,
escrows, market makers, DePIN apps, auctions, retail payment platforms, and
more—are referred to as **Solana apps**.
```

Comment (mikemaccana) [reply content/courses/onchain-development/intro-to-onchain.md:30]: Technically localnet isn't a cluster.

File: content/courses/onchain-development/intro-to-onchain.md [hunk 2]
```diff
@@ -11,133 +11,192 @@ description:
...
 
-- sending incoming instructions to the right instruction handlers
-- deserializing data from incoming transactions
-- checking the accounts provided with incoming instructions, for example, to
-  check that certain accounts are of a particular type, or are distinct from
-  other accounts.
+- Routing incoming instructions to the appropriate instruction handlers.
+- Deserializing data from incoming transactions.
+- Validating accounts provided with instructions, such as ensuring that certain
+  accounts are of a specific type or distinct from other accounts.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:40]:
```suggestion
- Deserializing data from incoming instructions.
- Validating accounts provided with instructions, such as ensuring that certain
  accounts have signed the transaction, are of a specific type, or are distinct from other accounts.
```

File: content/courses/onchain-development/intro-to-onchain.md [hunk 3]
```diff
@@ -11,133 +11,192 @@ description:
...
+#### Examples
+
+Here's a table illustrating various use cases for PDAs:
+
+| Use Case          | Seeds                          | PDA (Key)       | Value (Data Stored)                        |
+| ----------------- | ------------------------------ | --------------- | ------------------------------------------ |
+| Exchange Rate     | `["USD", "AUD"]`               | Derived address | Current USD to AUD exchange rate           |
+| User Relationship | `[user1_wallet, user2_wallet]` | Derived address | Relationship data (e.g., friends, blocked) |
+| Movie Review      | `[reviewer_wallet, "titanic"]` | Derived address | Review text, rating, timestamp             |
+| Global Config     | `["config"]`                   | Derived address | Program-wide settings                      |
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:124]: Not sure if I mentioned it before but I love this table, and will use it in our other work.

File: content/courses/onchain-development/intro-to-onchain.md [hunk 4]
```diff
@@ -11,133 +11,192 @@ description:
...
+  PROGRAM_ID,
+);
 ```
 
 ### There are multiple ways to build onchain, but we recommend Anchor
 
 You currently have two options for onchain program development:
 
-- We recommend new onchain programmers
+- We recommend new developers start with
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:190]:
```suggestion
- We recommend new developers 
```

File: content/courses/onchain-development/intro-to-onchain.md [hunk 5]
```diff
@@ -11,133 +11,192 @@ description:
...
-- There's also a separate
-  [native onchain program development](/content/courses/native-onchain-development)
-  course.
+  Anchor's defaults make it easy to create secure programs.
+- There is also a course on
+  [native onchain program development](/content/courses/native-onchain-development).
 
-Whichever way you pick, Solana Foundation maintains
+Whichever approach you choose, the Solana Foundation provides
 [examples in both languages](https://github.com/solana-developers/program-examples),
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:197]:
```suggestion
[examples in both Anchor and native Rust](https://github.com/solana-developers/program-examples),
```

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Great work, again, particularly on the PDA example just needs some small issues fixed. Note @0xCipherCoder you haven't won this one - there was already an update we've accepted as the winner - however if you fix the conflicts we'd love to get this work into the lesson.

File: content/courses/onchain-development/intro-to-onchain.md (hunk 3 above)
Comment (0xCipherCoder) [reply content/courses/onchain-development/intro-to-onchain.md:153]: Thanks! @mikemaccana I appreciate it.

File: content/courses/onchain-development/intro-to-onchain.md (hunk 1 above)
Comment (0xCipherCoder) [reply content/courses/onchain-development/intro-to-onchain.md]: I have merged the main branch and removed localnet from the list.

File: content/courses/onchain-development/intro-to-onchain.md (hunk 2 above)
Comment (0xCipherCoder) [reply content/courses/onchain-development/intro-to-onchain.md]: I have merged the main branch and incorporated suggestions accordingly.

File: content/courses/onchain-development/intro-to-onchain.md (hunk 4 above)
Comment (0xCipherCoder) [reply content/courses/onchain-development/intro-to-onchain.md]: Resolved as a suggestion.

File: content/courses/onchain-development/intro-to-onchain.md (hunk 5 above)
Comment (0xCipherCoder) [reply content/courses/onchain-development/intro-to-onchain.md]: I have updated the link text as per a suggestion.

**Comment by 0xCipherCoder** [comment]
> Great work, again, particularly on the PDA example just needs some small issues fixed. Note @0xCipherCoder you haven't won this one - there was already an update we've accepted as the winner - however if you fix the conflicts we'd love to get this work into the lesson.

Thanks! @mikemaccana I have merged the main branch and resolved the conflict along with comments. I have tried to keep the best of both. Please review it once and let me know if any additional changes are required.

**Comment by mikemaccana** [comment]
@nickfrosty sure you don't want to merge it? It's good work.

**Comment by nickfrosty** [comment]
To me, this reads as the same info with no meaningful change

**Pull request closed without merging**